Core/COEditingContext.m

```diff
@@ -412,8 +416,23 @@ - (void)updateCrossPersistentRootReferencesToPersistentRoot: (COPersistentRoot *
 {
 	NSParameterAssert(aPersistentRoot != nil);
 	// NOTE: -delete/undeleteBranch: enforce !aBranch.isCurrentBranch already.
-	NSParameterAssert(aBranch == nil
-		|| (!aBranch.isCurrentBranch && aBranch.persistentRoot == aPersistentRoot));
+	// NOTE: This is used by lazy loading now, and the following assertion no longer holds.
+	// Is this a problem?
+	//NSParameterAssert(aBranch == nil
+	//	|| (!aBranch.isCurrentBranch && aBranch.persistentRoot == aPersistentRoot));
+	
+	// See documentation above
+	if (isDeletion)
+	{
+		// TODO: For an uncommitted persistent root and branch, could be better if
```

Comment:
Cool.. the hack was necessary until your commit bec8cb9c2e89d850f363fb88fb0cd2d3de915bc6
(-testTargetPersistentRootDeletion at TestOrderedRelationship.m:268 would fail without the hack).